tests/plugins/lti_consumer/test_forms.py

```diff
@@ -111,3 +120,127 @@ def test_forms_lti_consumer_clean_valid(self):
                 self.assertDictEqual(form.errors, {})
                 instance = form.save()
                 self.assertIsNotNone(instance.url)
+
+    @override_settings(RICHIE_LTI_PROVIDERS=get_lti_settings())
+    @mock.patch.object(
+        exrex, "getone", return_value="http://localhost:8060/lti/videos/1234abcd-1"
+    )
+    def test_forms_lti_consumer_url_regex_match(self, _mock_getone):
+        """
+        The url field should match the regex url if a predefined LTI provider is
+        used and has a regex url.
+        """
+        data = {"lti_provider_id": "lti_provider_test", "url": "http://invalid.com"}
+        form = LTIConsumerForm(data=data)
+        self.assertFalse(form.is_valid())
+        self.assertDictEqual(
+            form.errors,
+            {
+                "url": [
+                    (
+                        "The url is not valid for this provider. "
+                        'It should be of the form "http://localhost:8060/lti/videos/1234abcd-1".'
+                    )
+                ]
+            },
+        )
+
+    @override_settings(RICHIE_LTI_PROVIDERS=get_lti_settings(is_regex=False))
+    def test_forms_lti_consumer_url_not_regex_included(self):
+        """
+        The url field should include the provider's base url if a predefined LTI provider
+        is used and has a regex url.
+        """
+        data = {"lti_provider_id": "lti_provider_test", "url": "http://invalid.com"}
+        form = LTIConsumerForm(data=data)
+        self.assertFalse(form.is_valid())
+        self.assertDictEqual(
+            form.errors,
+            {
+                "url": [
+                    (
+                        "The url is not valid for this provider. "
+                        'It should start with "http://localhost:8060/lti/videos/".'
+                    )
+                ]
+            },
+        )
+
+    @override_settings(RICHIE_LTI_PROVIDERS=get_lti_settings())
+    def test_forms_lti_consumer_reset_credentials(self):
+        """
+        The "oauth_consumer_key" and "shared_secret" fields should be reset when a value
+        is set for the "lti_provider_id" field.
+        """
+        lti_consumer = LTIConsumerFactory(
+            lti_provider_id=None,
+            oauth_consumer_key="OauthConsumerKey",
+            shared_secret="SharedSecret",
+        )
+        self.assertIsNotNone(lti_consumer.oauth_consumer_key)
+        self.assertIsNotNone(lti_consumer.shared_secret)
+
```

Comment:
```suggestion
lti_consumer = LTIConsumerFactory(
            lti_provider_id=None,
            oauth_consumer_key="OauthConsumerKey",
            shared_secret="SharedSecret",
            url= "http://localhost:8060/lti/videos/166d465f-f",
            
        )
        
        
        # in your factory it has been set so checking that is as we don't trust the factory
        # self.assertIsNotNone(lti_consumer.oauth_consumer_key)
        # self.assertIsNotNone(lti_consumer.shared_secret)
        data = LTIConsumerForm(instance=lti_consumer).initial
        data["form_shared_secret"] = "%%shared_secret_placeholder%%"
        form = LTIConsumerForm(instance=lti_consumer, data=data)
        form.is_valid()
        self.assertTrue(form.is_valid())
        lti_consumer.refresh_from_db()
        self.assertIsNotNone(lti_consumer.oauth_consumer_key)
        self.assertIsNotNone(lti_consumer.shared_secret)
        
        data = LTIConsumerForm(instance=lti_consumer).initial
        data.update(
            {
                "lti_provider_id": "lti_provider_test",
          #      "url": "http://localhost:8060/lti/videos/166d465f-f",
            }
        )
```

Submitting the form with a initial lti_provider_id at None was to me a reverse test that is more accurate than testing the factory. 
This way we're still testing that assertIsNone after testing assertIsNotNone. WDYT ?

Reply:
@carofun Ok. I can do that! :+1: